client/sniffer/src/main/java/org/elasticsearch/client/sniff/Sniffer.java

```diff
@@ -151,4 +155,27 @@ synchronized void shutdown() {
     public static SnifferBuilder builder(RestClient restClient) {
         return new SnifferBuilder(restClient);
     }
+
+    private static class SnifferThreadFactory implements ThreadFactory {
+
+        final ThreadGroup group;
+        final AtomicInteger threadNumber = new AtomicInteger(1);
+        final String namePrefix;
+
+        private SnifferThreadFactory(String namePrefix) {
+            this.namePrefix = namePrefix;
+            SecurityManager s = System.getSecurityManager();
+            group = (s != null) ? s.getThreadGroup() :
+                    Thread.currentThread().getThreadGroup();
+        }
+
+        @Override
+        public Thread newThread(Runnable r) {
```

Comment:
I think we should do this in a `AccessController.doPrivileged` block since we need a special permission to modify threadgroup. It's good to respect that since we use this in reindex etc. as well

Reply:
I got rid of this whole topic by just calling Executors.defaultFactory().newThread(Runnable). Somebody copied the code from java.util.concurrent.Executors.DefaultThreadFactory#DefaultThreadFactory to EsExecutors.